packages/react-three-create/src/integrations/triplex.ts

```diff
@@ -0,0 +1,215 @@
+import type { Generator } from '../index.js'
+import { unique } from '../lib/array.js'
+
+export type GenerateTriplexOptions = {} | boolean
+
+export type PropValue = {
+  declaredPropDefaultValue: unknown
+  declaredPropName: string
+  declaredPropType: string
+  propName: string
+  propValue: string
+}
+
+export type ProviderDefinition = Record<
+  string,
+  | {
+      component: string
+      type: 'wrapped-jsx'
+      import: string
+      props?: PropValue[]
+    }
+  | {
+      code: string
+      type: 'inline-jsx'
+      import: string
+      props?: PropValue[]
+    }
+  | {
+      code: string
+      type: 'layout-effect'
+      import: string
+      props?: PropValue[]
+    }
+>
+
+function generateProvidersModule(generator: Generator): string {
+  const canvasProviders: (keyof typeof providerDefs)[] = []
+  const globalProviders: (keyof typeof providerDefs)[] = []
+  const providerDefs: ProviderDefinition = {
+    uikit: {
+      type: 'layout-effect',
+      props: [
+        {
+          declaredPropDefaultValue: '"light"',
+          declaredPropName: 'colorMode',
+          propName: 'colorMode',
+          propValue: 'colorMode',
+          declaredPropType: '"light" | "dark"',
+        },
+      ],
+      code: `
+        setPreferredColorScheme(colorMode);
+      `,
+      import: 'import { setPreferredColorScheme } from "@react-three/uikit"',
+    },
+    rapier: {
+      component: 'Physics',
+      type: 'wrapped-jsx',
+      import: 'import { Physics } from "@react-three/rapier";',
+      props: [
+        {
+          declaredPropDefaultValue: false,
+          declaredPropName: 'physicsEnabled',
+          propName: 'paused',
+          propValue: '!physicsEnabled',
+          declaredPropType: 'boolean',
+        },
+        {
+          declaredPropDefaultValue: true,
+          declaredPropName: 'debugPhysics',
+          propName: 'debug',
+          propValue: 'debugPhysics',
+          declaredPropType: 'boolean',
+        },
+      ],
+    },
+    postprocessing: {
+      type: 'inline-jsx',
+      code: `
+        <EffectComposer enabled={postProcessingEnabled}>
+          <DepthOfField
+            focusDistance={0}
+            focalLength={0.02}
+            bokehScale={2}
+            height={480}
+          />
+          <Bloom luminanceThreshold={0} luminanceSmoothing={0.9} height={300} />
+        </EffectComposer>
+      `,
+      import: 'import { Bloom, DepthOfField, EffectComposer } from "@react-three/postprocessing";',
+      props: [
+        {
+          declaredPropDefaultValue: true,
+          declaredPropName: 'postProcessingEnabled',
+          propName: 'enabled',
+          propValue: 'postProcessingEnabled',
+          declaredPropType: 'boolean',
+        },
+      ],
+    },
+  }
+
+  if (!!generator.options.rapier) {
+    canvasProviders.push('rapier')
+  }
+
+  if (!!generator.options.postprocessing && !generator.options.xr) {
```

Comment:
@bbohlender I got stung with this where I wasn't checking for xr. Pros / cons I know but if integrations could mutate the options and then pass it on to other integrations would make remove some mental gymnastics of knowing if something is enabled or not (as XR could disable post processing, and then every subsequent integration assumes it's disabled).

But of course mutating can make it confusing who's doing what. Will just leave this though 😁.